core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt

```diff
@@ -459,11 +475,26 @@ class RawInfraImpl(
         return routeDescriptors[route].name
     }
 
+    override fun getRouteFromName(name: String): RouteId {
+        return routes.firstOrNull { getRouteName(it) == name } ?: throw OSRDError(ErrorType.UnknownRoute)
+    }
```

Comment:
That's an expensive call, we'd probably need a map there.

There's also a method in `fr.sncf.osrd.Helpers` that is roughly a duplicate and should be deleted.